src/types/error_parser.ts

```diff
@@ -0,0 +1,67 @@
+import * as _ from "lodash";
+
+import { Logging } from "./logging";
+import { DebtKernelError } from "./debt_kernel_error";
+import { RepaymentRouterError } from "./repayment_router_error";
+
+export interface ContractAddressBook {
+    debtKernel: string;
+    repaymentRouter: string;
+}
+
+enum Origin {
+    DebtKernel,
+    RepaymentRouter,
+}
+
+export class ErrorParser {
+    constructor(private contractAddressBook: ContractAddressBook) {
+        this.contractAddressBook = contractAddressBook;
+    }
+
+    public parseDecodedLogs = (logs: any[]): string[] => {
```

Comment:
I'd prefer if we kept the syntax we use to define function signatures consistent across the codebase, i.e. these signatures should look more like:

```javascript
public parseDecodedLogs(logs: any[]): string[] {
...
}
```

Reply:
I'm totally with you, except that this is where typescript can't solve the underlying crap that is javascript. I need to use arrow functions here to ensure that `this` is being scoped properly.

See: https://medium.freecodecamp.org/learn-es6-the-dope-way-part-ii-arrow-functions-and-the-this-keyword-381ac7a32881

Reply:
Okay, fair enough :) . I'm sufficiently convinced this is cleaner than doing `.bind(this)` in the constructor.

Reply:
+1